src/algebraic_geometry/open_immersion.lean

```diff
@@ -1634,21 +1634,60 @@ LocallyRingedSpace.is_open_immersion.lift_uniq f g H' l hl
   hom_inv_id' := by { rw ← cancel_mono f, simp },
   inv_hom_id' := by { rw ← cancel_mono g, simp } }
 
+/-- The functor `opens X ⥤ opens Y` associated with an open immersion `f : X ⟶ Y`. -/
+abbreviation _root_.algebraic_geometry.Scheme.hom.opens_functor {X Y : Scheme} (f : X ⟶ Y)
+  [H : is_open_immersion f] :
+  opens X.carrier ⥤ opens Y.carrier :=
+H.open_functor
+
+/-- The isomorphism `Γ(X, U) ⟶ Γ(Y, f(U))` induced by an open immersion `f : X ⟶ Y`. -/
+def _root_.algebraic_geometry.Scheme.hom.inv_app {X Y : Scheme} (f : X ⟶ Y)
+  [H : is_open_immersion f] (U) :
+  X.presheaf.obj (op U) ⟶ Y.presheaf.obj (op (f.opens_functor.obj U)) :=
+H.inv_app U
+
+lemma app_eq_inv_app_app_of_comp_eq_aux {X Y U : Scheme} (f : Y ⟶ U) (g : U ⟶ X)
+  (fg = f ≫ g) [h : is_open_immersion g] (V : opens U.carrier) :
+  (opens.map f.1.base).obj V = (opens.map fg.1.base).obj (g.opens_functor.obj V) :=
+begin
+  subst H,
```

Comment:
Please give the unnamed hyp a name. Or do you think we should change some of the guidelines?

Reply:
Does the guideline currently forbid this? The current form should be acceptable IMO as it is predictably named `H`, and it looks more succinct than `(fg : Y ⟶ X) (H : fg = f ≫ g)`. But I'm perfectly fine in changing it if people prefer the latter.

Reply:
Let's check with the mathport experts, what they think.
cc @digama0 @semorrison

Reply:
Currently, mathport will fail to produce correct code when you use this style:
```lean
theorem foo (x = 1) {P : nat → Prop} (h : P 1) : P x :=
begin
  subst H,
  exact h,
end
```
output:
```lean
theorem foo (x) (_ : x = 1) {P : Nat → Prop} (h : P 1) : P x := by
  subst H
  exact h
```